### PR TITLE
Object Tags

### PR DESCRIPTION
**Pull request by elliottomlinson**
[Tabletop Simulator API: Object Tag Functions](https://api.tabletopsimulator.com/object/#tag-functions)

Having objects tagged by their type (i.e. character figurine, map tile, stated token, etc.) and set (i.e. character folder, background area) should simplify scripting any object-handling features we implement in the future.

### CODE REVIEW

**Comment by elliottomlinson** [comment]
**Object Tags for Dynamic Item Resizing**

 We aim to design RPCG systems to be intuitive, so our players focus on the game, not the rules. As items/objects are one of the higher-granularity elements of _Cardmaster_, it has been difficult to create robust systems for handling them without restricting the players' agency within the world or interrupting the flow of gameplay with menus or rules.

Items are cards in our game. For each session (i.e. folder in assets/campaign), each image in the sets in `/item` becomes a card. (e.g. [colander](https://github.com/elliottomlinson/rpcg/blob/master/assets/campaign/03%20-%20Angel's%20Wings%20Redux/item/kitchen/Colander.png?raw=true) is a card in session 3's kitchen set). Cards in the same set are imported together as one object, and share backs (e.g. [kitchen](https://github.com/elliottomlinson/rpcg/blob/master/assets/campaign/03%20-%20Angel's%20Wings%20Redux/item/back/kitchen.png)) to simplify keeping track of them in-game. Once the items are placed on the map it is inherently obvious what the basics of interacting with them looks like - until now the problem is that the cards have had to be shrunk to fit on the map or dealt with another less-convenient way.

Now, the "item" tag that's now automatically applied to item cards on import flags them to be shrunk to half-size when placed on the map, allowing them to be clear in the hand and when picked up, and allowing them to be placed on the map as player-interactive decoration when setting up the session. For this to work, we must also create a scripting zone that covers the map and add the following script to Global.ttslua

```lua
function onObjectEnterScriptingZone(zone, obj)
    if obj.hasTag("item")
    then
      obj.setScale({0.5, 0.5, 0.5})
    end
end

function onObjectLeaveScriptingZone(zone, obj)
  if obj.hasTag("item")
  then
    obj.setScale({0.5, 0.5, 0.5})
  end
end
```

File: src/import-bg.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@
       url:url,
       posZ:posZ,
       posX:posX,
+      tag:set,
```
Comment (DuncanUszkay1) [inline src/import-bg.rb:32]: maybe try following the style guide you animal
```suggestion
      tag: set,
```

**Review by DuncanUszkay1 — APPROVED** [review]
It won't work but I'll allow you to merge it just to teach you a lesson about the pain of improper style usage

**Comment by DuncanUszkay1** [comment]
:shipit:

**Pull request merged**